### PR TITLE
Fix some redundancies and use Kotlin Deprecated annotation

### PR DESCRIPTION
**Pull request by hzsweers**
For redundancies: Removed redundant `PUBLIC` KModfier and Override/Nullable annotations
For deprecated annotations: Defaulted to ERROR as the level. Also fixed the doc ref

### CODE REVIEW

**Comment by hzsweers** [comment]
CC @asiri-uber

File: artist-traits-rx/src/main/kotlin/com/uber/artist/traits/rx/KotlinApiHelper.kt [hunk 1]
```diff
@@ -165,7 +163,6 @@ fun addRxBindingApiForSettable(type: TypeSpec.Builder, api: KotlinSettableApi, i
             .addKdoc(rxBindingMethodDoc)
             .apply {
                 if (api.isUViewOverride) {
-                    addAnnotation(Override::class.java)
```
Comment (jbarr21) [inline artist-traits-rx/src/main/kotlin/com/uber/artist/traits/rx/KotlinApiHelper.kt]: why is this removed? will kotlinpoet/kotlin add the Override annotation if you add that modifier?

File: artist-core/src/main/kotlin/com/uber/artist/KotlinArtistCodeGenerator.kt [hunk 2]
```diff
@@ -74,7 +74,7 @@ class KotlinArtistCodeGenerator : ArtistCodeGenerator<FileSpec, TypeSpec.Builder
   override fun createInitBuilderFor(stencil: KotlinViewStencil, type: TypeSpec.Builder): FunSpec.Builder {
     return FunSpec.builder("init")
         .addAnnotation(KotlinTypeNames.Annotations.CallSuper)
-        .addModifiers(KModifier.OPEN, KModifier.PROTECTED)
+        .addModifiers(KModifier.PROTECTED, KModifier.OPEN)
```
Comment (jbarr21) [inline artist-core/src/main/kotlin/com/uber/artist/KotlinArtistCodeGenerator.kt]: nice, thx for filing issue, too. how did you find this? ktlint, kotlin warning, or ide inspection?

File: artist-core/src/main/kotlin/com/uber/artist/KotlinArtistCodeGenerator.kt [hunk 3]
```diff
@@ -53,7 +53,7 @@ class KotlinArtistCodeGenerator : ArtistCodeGenerator<FileSpec, TypeSpec.Builder
   override fun generateTypeSpecFor(stencil: KotlinViewStencil, rPackageName: String, traitMap: Map<Class<out KotlinTrait>, KotlinTrait>, superinterfaceClassName: String?): TypeSpec.Builder {
     val rClass = ClassName(rPackageName, "R")
     val typeBuilder = TypeSpec.classBuilder(stencil.name())
-        .addModifiers(KModifier.PUBLIC, KModifier.OPEN)
```
Comment (jbarr21) [inline artist-core/src/main/kotlin/com/uber/artist/KotlinArtistCodeGenerator.kt]: what is redundant? does kotlinpoet assume open methods should be public?

Comment (hzsweers) [reply artist-core/src/main/kotlin/com/uber/artist/KotlinArtistCodeGenerator.kt]: public is the default in kotlin

File: artist-core/src/main/kotlin/com/uber/artist/KotlinArtistCodeGenerator.kt (hunk 2 above)
Comment (hzsweers) [reply artist-core/src/main/kotlin/com/uber/artist/KotlinArtistCodeGenerator.kt]: Just noticed eyeballing it. KotlinPoet is supposed to do this already though, so something else is going on

File: artist-traits-rx/src/main/kotlin/com/uber/artist/traits/rx/KotlinApiHelper.kt (hunk 1 above)
Comment (hzsweers) [reply artist-traits-rx/src/main/kotlin/com/uber/artist/traits/rx/KotlinApiHelper.kt]: kotlin uses the `override` modifier, the annotation is redundant/not idiomatic

**Comment by hzsweers** [comment]
@jbarr added a few more bits, will wait for you to do another pass before merging

**Comment by jbarr21** [comment]
👍 lgtm
